### DISCUSSION TITLE
Tutorials and BlogPosts

### DISCUSSION THREAD

**Discussion by Borda** [Lightning-AI/pytorch-lightning#7511] [General]
## Blog posts

* [From PyTorch to PyTorch Lightning — A gentle introduction](https://towardsdatascience.com/from-pytorch-to-pytorch-lightning-a-gentle-introduction-b371b7caaf09)
 PyTorch is extremely easy to use to build complex AI models. But once the research gets complicated and things like multi-GPU training, 16-bit precision and TPU training get mixed in, users are likely to introduce bugs.
 PyTorch Lightning solves exactly this problem. Lightning structures your PyTorch code so it can abstract the details of training. This makes AI research scalable and fast to iterate on.
* [36 Ways Pytorch Lightning Can Supercharge Your AI Research](https://towardsdatascience.com/supercharge-your-ai-research-with-pytorch-lightning-337948a99eec)
* [Pytorch Lightning vs PyTorch Ignite vs Fast.ai](https://towardsdatascience.com/pytorch-lightning-vs-pytorch-ignite-vs-fast-ai-61dc7480ad8a)
 PyTorch-lightning is a recently released library which is a Kera-like ML library for PyTorch. It leaves core training and validation logic to you and automates the rest. (BTW, by Keras I mean no boilerplate, not overly-simplified).
* [Trivial Multi-Node Training With Pytorch-Lightning](https://towardsdatascience.com/trivial-multi-node-training-with-pytorch-lightning-ff75dfb809bd)
* [Converting From Keras To PyTorch Lightning](https://towardsdatascience.com/converting-from-keras-to-pytorch-lightning-be40326d7b7d) -
 In this tutorial, we’ll convert a Keras model into a PyTorch Lightning model to add another capability to your deep-learning ninja skills.
* [Simplify And Scale Your PyTorch Code With PyTorch Lighting](https://towardsdatascience.com/how-to-refactor-your-pytorch-code-to-get-these-42-benefits-of-pytorch-lighting-6fdd0dc97538) -
 PyTorch Lightning is a very lightweight wrapper on PyTorch which is more like a coding standard than a framework. If you refactor your PyTorch code into the Lightning format you get the bells and whistles of top research teams without all the work. The format allows you to get rid of a ton of boilerplate code and makes it highly reproducible.
* [Torchbearer Joins PyTorch Lightning](https://medium.com/pytorch/pytorch-frameworks-unite-torchbearer-joins-pytorch-lightning-c588e1e68c98)
* [PyTorch Lightning 0.7.1 Release and Venture Funding](https://medium.com/pytorch/pytorch-lightning-0-7-1-release-and-venture-funding-dd12b2e75fb3) -
 The 0.7.1 release signals a new level of framework maturity. With major API changes behind us, this release paves the way to the major 1.0 milestone we aim to reach this year.
* [En-Lightning Reinforcement Learning](https://towardsdatascience.com/en-lightning-reinforcement-learning-a155c217c3de) -
 This article looks at using PyTorch Lightning for the exciting domain of Reinforcement Learning (RL). Here we are going to build a standard Deep Q Network (DQN) model using the classic CartPole gym environment to illustrate how to start using Lightning to build your RL models.
* [Custom classifier on top of BERT-like Language Model - guide](https://zablo.net/blog/post/custom-classifier-on-bert-model-guide-polemo2-sentiment-analysis)
* [Training Neural Networks for Leela Zero With PyTorch](https://towardsdatascience.com/training-neural-networks-for-leela-zero-using-pytorch-and-pytorch-lightning-bbf588683065) -
 A simple training pipeline for Leela Zero implemented with PyTorch, PyTorch Lightning and Hydra
* [Log PyTorchLightning metrics to Neptune](https://docs.neptune.ai/integrations/pytorch_lightning.html)
* [Use Pytorch Lightning with Weights & Biases](https://app.wandb.ai/cayush/pytorchlightning/reports/Use-Pytorch-Lightning-with-Weights-%26-Biases--Vmlldzo2NjQ1Mw)
* [Using Optuna to Optimize PyTorch Lightning Hyperparameters](https://medium.com/optuna/using-optuna-to-optimize-pytorch-lightning-hyperparameters-d9e04a481585) -
 Optuna is a hyperparameter optimization framework applicable to machine learning frameworks and black-box optimization solvers. PyTorch Lightning provides a lightweight PyTorch wrapper for better scaling with less code. Combining the two of them allows for automatic tuning of hyperparameters to find the best performing models.
* [How to Keep Track of PyTorch Lightning Experiments with Neptune](https://neptune.ai/blog/pytorch-lightning-neptune-integration)
* [How I built a Face Mask Detector for COVID-19 using PyTorch Lightning](https://towardsdatascience.com/how-i-built-a-face-mask-detector-for-covid-19-using-pytorch-lightning-67eb3752fd61)
* [Getting Started with PyTorch Lightning](https://www.learnopencv.com/getting-started-with-pytorch-lightning)
* [PyTorch Multi-GPU Metrics Library and More in PyTorch Lightning 0.8.1](https://medium.com/pytorch/pytorch-multi-gpu-metrics-and-more-in-pytorch-lightning-0-8-1-b7cadd04893e) - 
 Today we released 0.8.1 which is a major milestone for PyTorch Lightning. With incredible user adoption and growth, we’re continuing to build tools to easily do AI research.
* [PyTorch Lightning: Metrics](https://medium.com/pytorch/pytorch-lightning-metrics-35cb5ab31857)
 With PyTorch Lightning 0.8.1 we added a feature that has been requested many times by our community: Metrics. This feature is designed to be used with PyTorch Lightning as well as with any other PyTorch based code. In this blog post, we’ll explain what Metrics is and how you can get started.
* [fastai and PyTorch Lightning are democratising AI](https://towardsdatascience.com/fastai-and-pytorch-lightning-are-democratising-ai-92318ff224b1)
* [How To Tag Any Image Using Deep Learning](https://towardsdatascience.com/how-to-tag-any-image-using-deep-learning-84a0dc2e03c2)
* [Distributed Deep Learning with Ansible, AWS and Pytorch Lightning. Part 1](https://towardsdatascience.com/distributed-deep-learning-with-ansible-aws-and-pytorch-lightning-part-1-91d500e1039f)
* [Use Pytorch Lightning to Decouple Science science code from engineering code](https://medium.com/analytics-vidhya/use-pytorch-lightning-to-decouple-science-science-code-from-engineering-code-1e6f36cf6a1a)
* [Productive NLP Experimentation with Python using Pytorch Lightning and Torchtext](https://towardsdatascience.com/productive-nlp-experimentation-with-python-using-pytorch-lightning-and-torchtext-256a4d00a26b)
* [Video Prediction using Deep Learning and PyTorch-lightning](https://towardsdatascience.com/video-prediction-using-convlstm-with-pytorch-lightning-27b195fd21a2)
 In this guide, I will show you how to code a Convolutional Long Short-Term Memory (ConvLSTM) using an autoencoder (seq2seq) architecture for frame prediction using the MovingMNIST dataset (but custom datasets can also easily be integrated).
* [Automate Your Neural Network Training With PyTorch Lightning](https://medium.com/swlh/automate-your-neural-network-training-with-pytorch-lightning-1d7a981322d1)
 PyTorch Lightning will automate your neural network training while staying your code simple, clean, and flexible. If you’re a researcher you will love this!
* [TensorBoard with PyTorch Lightning](https://www.learnopencv.com/tensorboard-with-pytorch-lightning)
* [Effective testing for machine learning systems](https://www.jeremyjordan.me/testing-ml/amp/)
 Working as a core maintainer for PyTorch Lightning, I've grown a strong appreciation for the value of tests in software development. As I've been spinning up a new project at work, I've been spending a fair amount of time thinking about how we should test machine learning systems.
* [Keeping Up with PyTorch Lightning and Hydra](https://medium.com/@peterkeunwoo/keeping-up-with-pytorch-lightning-and-hydra-31e8ed70b2cc)
 Last week, PyTorch Lightning 0.9.0 and Hydra’s fourth release candidate for 1.0.0 were released with a choke-full of new features and mostly final APIs. I thought it’d be a good time for me to revisit my side project Leela Zero PyTorch to see how these new versions can be integrated into it. In this post, I’ll talk about some of the new features of the two libraries, and how they helped Leela Zero PyTorch.
* [EfficientDet Meets Pytorch-Lightning](https://medium.com/@YassineAlouini/efficientdet-meets-pytroch-lightning-6f1dcf3b73bd)
 Welcome to this beginner friendly guide to object detection using EfficientDet. Similarly to what I have done in the NLP guide (check it here if you haven’t yet already), there will be a mix of theory, practice, and an application to the global wheat competition dataset.
* [Automatically finding good learning rate for your network with PyTorch Lightning](https://enjoymachinelearning.com/posts/find-lr-pytorch-lightning)
 Among of all hyperparameters used in machine learning, learning rate is probably the very first one you hear about. It may also the one that you start tuning in the first place. You can find the right value with a bit of hyper parameter optimization, running tons of training sessions or you can let tools do it, much faster. Nowadays, many libraries implement LR Finder or “LR Range Test”.
* [3 Simple Tricks That Will Change the Way You Debug PyTorch](https://medium.com/@adrian.waelchli/3-simple-tricks-that-will-change-the-way-you-debug-pytorch-5c940aa68b03)
 Every Deep Learning project is different. No matter how much experience you bring with you, there will always be new challenges and unexpected behavior you will struggle with. The skill- and mindset that you bring to the project will determine how quickly you discover and adapt to the obstacles that stand in the way of success.
